dts/bindings/display/ftdi,ft800.yaml

```diff
@@ -1,7 +1,28 @@
 # Copyright (c) 2020 Hubert Miś <hubert.mis@gmail.com>
 # SPDX-License-Identifier: Apache-2.0
 
-description: FTDI FT800 graphic controller
+description: |
+    FTDI FT800 graphic controller
+
+    - Single chip for Display, Audio and Resistive Touch control
+    - Up to 512 x 512 pixel resolution
+    - 262K colours (6 Red, 6 Green, 6 Blue)
+    - Anti-Aliasing
+    - 2-bit colour dither
+    - Programmable HSYNC/VSYNC timing
+    - Backlight control
+    - Resistive touch control
+    - Object oriented programming approach
+    - Embedded widgets to assist with creating complex images
+    - Mono audio output
+    - 64 in-built polyphonic tones
+    - Decode for mono 8-bit linear PCM, 4-bit ADPCM and μ-Law coding format at sampling frequencies from 8kHz to 48kHz
+    - SPI/I2C interface to host MCU
+    - 1.8V to 3.3V IO levels to host MCU
+    - 3V3 levels on RGB pins
+    - Extended operating temperature range -40°C to 85°C
+    - Available in QFN-48 package (RoHS compliant)
```

Comment:
In general I think bindings don't need to describe all hardware features (datasheets are really the place for that information), but rather what the drivers really support and sometimes example usages. For example, whether RGB pins support 3V3 is not relevant here.

Reply:
I agree that not all information from a datasheet does make sense here.
Without any guidelines which information may resp. may not be included in the dts description I just basically copied what the driver vendor considered the core / main features  of the hw driver (which varies a lot between them).
It was a fast approach without selecting.

I will not be able (time-wise) to dig into the code and document what of all the hw features the software driver supports. 

Background:
I had to search if Zephyr OS supports other display driver for low res OLED displays than the SSD1306.
The idea behind adding those info was that, with the docus search functionality it would be easier to find what me and others are looking for instead of getting a list of hw drivers and then visiting the vendors web page and download and read the datasheets.
Since I had to do that now for my search, I provided the info (basically unfiltered) here.

I suggest:
INs:
- resolution
- technology (OLED, LCD, TFT, etc.)
- color depth
- host interface (i2c etc.)
- addon features
  - audio
  - touch
  - backlight control
  - graphical features (anti-aliasing etc.)
  - (temperature) sensors
- [basically anything which has a functional relevance (also to the software driver)]

OUTs:
- operation / storage temperature ranges
- voltages
- clock rates
- packages

I actually see a benefit in having information also if the driver does not support it (yet, it must be (made) clear that in the dts chip description, the chip capabilities are meant and not the software drivers).
If someone finds the wanted product e.g. an OLED driver already integrated in Zephyr OS, even if only partial, she/he uses it and might add the "rest" (e.g. the temperature sensor) if needed.

Reply:
> I think Devicetree bindings should describe, in general, supported features by the driver implementing it

Sorry to throw a wrench in the works, but I disagree with @gmarull on this specific point.

Bindings should describe the hardware and its configuration. They should ideally be independent of the features supported by a particular driver.

(We slightly deviate from this ideal sometimes when we have driver-specific configuration values in devicetree properties. These should be named `zephyr-<prop-name>` to make this clear.)

Drivers themselves and their capabilities should be documented at a higher level. People try to stick driver-specific information in the bindings, but this is not the right place for it.

This will be important if we ever move over to sharing devicetree bindings with Linux, by adapting our tooling to accept the Linux YAML binding schema. We would want to be able to reuse upstream Linux bindings without having to edit them by adding additional information about the relevant Zephyr drivers' abilities.

The fact that we have our own bindings language is due to the desires of the implementer and former maintainer of the DT tooling, but this was a controversial decision and I'd like to revisit it someday. So from a layering perspective I am firm about trying to keep zephyr specifics out of bindings.

Reply:
> Sorry to throw a wrench in the works, but I disagree with @gmarull on this specific point.
> 
> Bindings should describe the hardware and its configuration. They should ideally be independent of the features supported by a particular driver.
> 

That’s more accurate, I agree and makes sense. Still, I think that we should document the supported properties or how they map to hardware features, maybe with some examples.

Converting descriptions to a pseudo-datasheet doesn’t sound like a good idea, though.

> This will be important if we ever move over to sharing devicetree bindings with Linux, by adapting our tooling to accept the Linux YAML binding schema. We would want to be able to reuse upstream Linux bindings without having to edit them by adding additional information about the relevant Zephyr drivers' abilities.

That will be nice, however, I think it will be difficult to share everything. Bindings definition doesn’t follow any particular rule, it’s in many cases driven by an implementation (a Linux driver), which may not necessarily align with Zephyr requirements.

Reply:
My intention was to make it easy for people searching for a hardware with certain features / attributes in the Zephyr OS documentation because they require it for their project and would like to know if Zephyr OS supports it (also if only partial) - ease of use.
This is in addition to sample descriptions.
I see the Zephyr OS documentation as some kind of database with also some key datasheet information of supported hardware in it.
@gmarull Where do you see such information to be stored in the Zepyhr docu? Or not at all, just leave it in the vendors datasheet?

What is key information I tried to specify (INs and OUTs) but open to be discussed. 
For clarification to the user I could add a title line "Features (of hardware device, not software driver)" and reduce the feature list to my proposal above.

But it seems a general decision. If consensus can not be found, we drop it.